Understanding survey data files (#23)
tidy-survey-r/tidy-survey-book: 03-understanding-survey-data-files.Rmd

In addition to survey design, the technical documentation may include other helpful information. Some technical documentation includes syntax for SAS, SUDAAN, Stata, and/or R, meaning we don't have to create this code from scratch. This saves us time and ensures our code is accurate.

### Questionnaires

A survey questionnaire is the series of questions asked to obtain information from survey respondents. A questionnaire may employ different types of questions, such as multiple-choice, open-ended, Likert scales, or ranking questions, to gather opinions, behaviors, or demographic data. The questionnaire may also randomize responses or include instructions to help the respondent understand the question. A survey may have one questionnaire or multiple, depending on its scale and scope.

> The questionnaire is an essential resource for understanding and interpreting the survey data, and should be used alongside any analysis. It provides details about each of the questions asked in the survey, such as question name, question wording, response options, question logic, randomizations, display specification, mode differences, and the universe (if only a subset of respondents were asked the question). 

Reviewer: skip logic instead of question logic

```suggestion
The questionnaire is an essential resource for understanding and interpreting the survey data, and should be used alongside any analysis. It provides details about each of the questions asked in the survey, such as question name, question wording, response options, skip logic, randomizations, display specification, mode differences, and the universe (if only a subset of respondents were asked the question). 
```

> Below, we show a question from the ANES 2020 questionnaire. This part shows a particular question's question name (`postvote_rvote`), description (Did R Vote?), full wording of the question and responses, response order, universe, question logic (if `vote_pre` = 0), and other specifications. It also includes the variable name, which can be used to link to the codebook.
> 
> ![](images/questionnaire-example.jpg)

Reviewer: I prefer images with captions and we need to add in alt-text. One way to do this:

````suggestion
Below in Figure \@ref(fig:que-examp), we show a question from the ANES 2020 questionnaire. This part shows a particular question's question name (`postvote_rvote`), description (Did R Vote?), full wording of the question and responses, response order, universe, question logic (if `vote_pre` = 0), and other specifications. It also includes the variable name, which can be used to link to the codebook.

```{r}
#| label: que-examp
#| echo: false
#| fig.cap: ANES 2020 Questionnaire Example
#| fig.alt: Question information about the variable postvote_rvote from ANES 2020 questionnaire Survey question, Universe, Logic, Web Spec, Response Order, and Released Variable are included. 

knitr::include_graphics(path="images/questionnaire-example.jpg")
```
````

### Codebooks

While a questionnaire provides information about the questions asked to respondents, the codebook explains how the survey data was coded and recorded. The codebook lists details such as variable names, variable label, variable meaning, codes for missing data, values labels, and value type (whether it is categorical or continuous, etc.). The codebook enables us to understand the variables and use them appropriately in our analysis.

Below, we show a question from the ANES 2020 codebook. This part shows a particular variable's name (`V202066`), question wording, value labels, universe, and associated survey question (`postvote_rvote`).

> ![](images/codebook-example.jpg)

Reviewer: Update this as similarly above to have caption and alt-text

### Errata

An erratum (singular) or errata (plural) is a document that lists errors found in a publication or dataset, such as a survey questionnaire. The purpose of an erratum is to correct or update mistakes or inaccuracies in the original document.

For example, if a survey questionnaire contains a question with an error, such as a typo or a confusing wording, an erratum would be created to provide a corrected version of the question. This helps to ensure the accuracy and reliability of the survey data and analysis.

Be sure to review these lists before conducting any analysis to ensure that the most accurate and up-to-date information is used in the analysis.

### Additional resources

Surveys may have additional resources, such as interviewer instructions or "show cards" held up to respondents to help them answer a question. Review the survey website to find out more about these resources.

## Working with missing data

> If a respondent does not reply to a survey question, we consider the data missing. Respondents may not respond to a question for a variety of reasons, such as not wanting to answer a particular question, not understanding the question, or simply forgetting to provide a response.

Reviewer: Additionally, it may have been skipped by design. Mention this here but will be handled in next subsection

```{r}
dat <- tibble::tribble(
  ~col1, ~col2, ~col3,
    "a",    "d",   "e",
    "b",    NA,    NA,
    "c",    NA,    "f"
  )

dat %>% 
  tidyr::drop_na()
```

However, if our data is not missing completely at random (MCAR), listwise deletion may produce biased estimates (there may be a pattern of respondents who do not respond to specific questions). In these circumstances, we should explore other options, such as multiple imputation or weighted estimation. See Allison (2002) for more detail.
<!-- this was referenced in the ANES user guide but maybe we should not include?-->

### Accounting for skip patterns

Questionnaires may also include skip patterns, in which certain questions are skipped based on the respondent's answers to earlier questions. For example, if a respondent answers "no" to a question on whether they voted in the last election, then they may be instructed to skip a series of questions related to that election.

> As with missing data, we need to account for skip patterns in our survey analysis to ensure unbiased and accurate population parameters. We can treat skipped questions as missing data. Or, we can also incorporate skip patterns into the analysis and run analysis that account for the conditional dependence between the skipped questions and the questions that were answered. The appropriate method depends on the nature and extent of the skip patterns in the survey, as well as the research questions and methodology.

Reviewer: ```suggestion
As with missing data, we need to account for skip patterns in our survey analysis to ensure unbiased and accurate population parameters. We can treat skipped questions as missing data. Or, we can also incorporate skip patterns into the analysis and run analysis that account for the conditional dependence between the skipped questions and the questions that were answered. The appropriate method depends on the nature and extent of the skip patterns in the survey, as well as the research questions and methodology. For example, if we wanted to know what proportion of eligible voters voted for a particular candidate, the denominator would be all who are eligible while if we wanted to know what proportion voted for a particular candidate among those who voted, the denominator would be those who voted. This changes whether the missing values are included or excluded in an analysis.
```

  count(state)
```

    ## # A tibble: 51 x 2
    ##    state     n
    ##    <dbl> <int>
    ##  1     1  2406
    ##  2     2  1289
    ##  3     4  1969
    ##  4     5  1988
    ##  5     6  9574
    ##  6     8  1365
    ##  7     9  1157
    ##  8    10  1285
    ##  9    11  1622
    ## 10    12  5055
    ## # ... with 41 more rows

``` r
# Confirm this is only November 2020
cps_state %>%
  count(HRMONTH, HRYEAR4)
```

    ## # A tibble: 1 x 3
    ##   HRMONTH HRYEAR4      n
    ##     <dbl>   <dbl>  <int>
    ## 1      11    2020 112037

``` r
# Voting age citizen population
targetpop <- cps_state %>%
  as_tibble() %>%
  filter(PRTAGE >= 18,
         PRCITSHP %in% (1:4)) %>%
  pull(PWSSWGT) %>%
  sum()
```

The target population in 2020 is 231,592,693. This information gives us what we need to create the post-election survey object with {srvyr}:

```{r}
#| eval: FALSE
library(tidyverse)
library(here)
library(srvyr)

anes <- read_rds(here("Data", "anes_2020.rds")) %>%
  mutate(Weight = V200010b / sum(V200010b) * 231592693)

anes_des <- anes %>%
  as_survey_design(
    weights = Weight,
    strata = V200010d,
    ids = V200010c,
    nest = TRUE
  )

summary(anes_des)

> ```

Reviewer: Is there any reason this code isn't actually run? I think we should and show that the value of targetpop=231592693

Author: I think it was because of not having the data in the folders and also not having a Census key :D I am adding both now.
